packages/react-components/src/components/trace-context-component.tsx

```diff
@@ -52,7 +52,14 @@ interface TraceContextState {
     backgroundTheme: string;
 }
 
-export class TraceContextComponent extends React.Component<TraceContextProps, TraceContextState> {
+interface UndoRedoFunctionality {
+    undoStack: bigint[][];
+    redoStack: bigint[][];
+    undo: () => void;
```

Comment:
Maybe add TraceContextComponent to this component as an argument...

Reply:
I tried to store TraceContextComponent in the stack and add TraceContextComponent as an argument, but it is passed by reference. As a result, no value can be changed. Instead, if undo function returns the viewRange, it is possible to do 'this.viewRange = undo()', and it passed four checks. I also pushed modified code and a new commit message.